core/DataAccess/Model.php

```diff
@@ -973,6 +973,28 @@ public function resetFailedArchivingJobs()
         return $query->rowCount();
     }
 
+    public function getRecordsContainedInArchives(Date $archiveStartDate, $idArchives, $requestedRecords)
+    {
+        $idArchives = array_map('intval', $idArchives);
+        $idArchives = implode(',', $idArchives);
+
+        $requestedRecords = is_string($requestedRecords) ? [$requestedRecords] : $requestedRecords;
+        $placeholders = Common::getSqlStringFieldsArray($requestedRecords);
+
+        $countSql = "SELECT DISTINCT name FROM %s WHERE idarchive IN ($idArchives) AND name IN ($placeholders) LIMIT " . count($requestedRecords);
+
+        $numericTable = ArchiveTableCreator::getNumericTable($archiveStartDate);
+        $blobTable = ArchiveTableCreator::getBlobTable($archiveStartDate);
+
+        $existingRecords = [];
+        foreach ([$numericTable, $blobTable] as $tableName) {
+            $sql = sprintf($countSql, $tableName);
+            $rows = Db::fetchAll($sql, $requestedRecords);
```

Comment:
maybe minor performance tweak for later.

If one of the record names starts with `nb_` for example, then we could start looking in the numeric table and if we find records, maybe we wouldn't need to search in the blob table? And the other way around if we search in the blob table for some records and find entries we wouldn't need to look in the other table. This assumes that people usually either call `getNumeric()` or `getBblob()` and don't mix up records.

Reply:
:+1: added it